modules/core/src/main/scala/sangria/execution/ValueCollector.scala

```diff
@@ -63,7 +64,14 @@ class ValueCollector[Ctx, Input](
             }
         }
 
-      val (errors, values) = res.partition(_._2.isLeft)
+      val (allErrors, values) = res.partition(_._2.isLeft)
+
+      // additionally reduce the number of errors
+      // although we limit the number of errors per input, it's still possible to exceed the limit if errors are combined.
+      val errors = errorsLimit match {
+        case Some(limit) => allErrors.take(limit)
+        case _ => allErrors
+      }
```

Comment:
it seems like there are other places where violations could be returned, but I'm not sure if they should be updated as well.